tests/platform_tests/api/test_fan_drawer.py

```diff
@@ -24,22 +24,27 @@
 STATUS_LED_COLOR_OFF = "off"
 
 
-class TestFan_drawer_DrawerApi(PlatformApiTestBase):
+class TestFanDrawerApi(PlatformApiTestBase):
 
     num_fan_drawers = None
+    fan_drawer_truth = None
 
     # This fixture would probably be better scoped at the class level, but
     # it relies on the platform_api_conn fixture, which is scoped at the function
     # level, so we must do the same here to prevent a scope mismatch.
 
     @pytest.fixture(scope="function", autouse=True)
-    def setup(self, platform_api_conn):
+    def setup(self, duthost, platform_api_conn):
         if self.num_fan_drawers is None:
             try:
                 self.num_fan_drawers = int(chassis.get_num_fan_drawers(platform_api_conn))
             except:
                 pytest.fail("num_fan_drawers is not an integer")
 
+        chassis_truth = duthost.facts.get('chassis', None)
+        if chassis_truth:
+            self.fan_drawer_truth = chassis_truth.get('fan_drawers', None)
```

Comment:
I think there should be an `else` clause here which logs a warning message stating that the no platform facts were found, so the test will not be as thorough.